### ISSUE TITLE
Wordforms are not taken into account anymore

### ISSUE THREAD

**Issue by ponceta** [geoadmin/service-sphinxsearch#203]
@gjn @ltclm @loicgasser Any idea what change we made in the last month(s) that could have caused this? New search? Streaming replication? Moving some things to chsdi?

(Same for stopwords.)

**Comment by gjn** [comment]
We should have had tests, which would have enabled us to detect when the change happened.

Anyway: I'm not sure, but did you verify that the wordforms exist in the proper places on all instances? Indices are always created on the instances.

**Comment by ponceta** [comment]
We had tests, just not all swisssearch indices types ... 

Wordforms files are correctly stored at each location.

**Comment by gjn** [comment]
We didn't have tests for wordforms specifically.

Anyway...no idea why they don't work anymore. Could you give an example of a query and what it should return?

**Comment by gjn** [comment]
Some wordforms work:

without wordforms: avenue de la gare gryon
with wordforms: boulevard de la gare gryon (or boul de la gare gryon)

**Comment by gjn** [comment]
I tried other wordforms and I think wordforms still work as expected.

The example with rte was maybe unfortunate. @ponceta Could you give me your exact use case?

**Comment by ponceta** [comment]
@gjn try Ch. des Prés-du-Dimanche as example in #150 I wish it would be unfortunate

boul is autocompleted that's why you find boulevard.

**Comment by gjn** [comment]
autocompleted? Where do we autocomplete?

Thanks for the example. I'll have a deeper look.

**Comment by ponceta** [comment]
It is not really autocomplete but if you search for "dima" you will have results like "dimanche"

**Comment by gjn** [comment]
But that doesn't explain that the entry `avenue de la gare gryon` can be found with `boul de la gare gryon`. This can only be explained with wordfroms, no? (note that `boule de la gare` (added `e`) does not find anything)

**Comment by ltclm** [comment]
how do you know that they are not taken into account,
at least the wordforms files are present on all sphinx instances
```bash
$ for host in localhost ip-10-220-4-39.eu-west-1.compute.internal ip-10-220-5-210.eu-west-1.compute.internal ip-10-220-6-242.eu-west-1.compute.internal; do echo $host; ssh $host ls /var/lib/sphinxsearch/data/index/wordforms_*; done
localhost
/var/lib/sphinxsearch/data/index/wordforms_main.txt

ip-10-220-4-39.eu-west-1.compute.internal
/var/lib/sphinxsearch/data/index/wordforms_main.txt

ip-10-220-5-210.eu-west-1.compute.internal
/var/lib/sphinxsearch/data/index/wordforms_main.txt

ip-10-220-6-242.eu-west-1.compute.internal
/var/lib/sphinxsearch/data/index/wordforms_main.txt
```

and the wordforms are part of the active indexes:

```bash
$ indextool --dumpconfig /var/lib/sphinxsearch/data/index/gg25.sph
Sphinx 2.1.6-id64-release (rel21-r4571)
Copyright (c) 2001-2014, Andrew Aksyonoff
Copyright (c) 2008-2014, Sphinx Technologies Inc (http://sphinxsearch.com)

dumping header file '/var/lib/sphinxsearch/data/index/gg25.sph'...

source $dump
{
        sql_query = SELECT id \
        , detail \
        , geom_quadindex \
        , feature_id \
        , detail \
        , label \
        , origin \
        , geom_quadindex \
        , geom_st_box2d \
        , rank \
        , x \
        , y \
        , lat \
        , lon \
        , num \
        FROM documents

        sql_attr_string = feature_id
        sql_attr_string = detail
        sql_attr_string = label
        sql_attr_string = origin
        sql_attr_string = geom_quadindex
        sql_attr_string = geom_st_box2d
        sql_attr_uint = rank
        sql_attr_float = x
        sql_attr_float = y
        sql_attr_float = lat
        sql_attr_float = lon
        sql_attr_uint = num
}

index $dump
{
        source = $dump
        path = $dump
        min_prefix_len = 1
        charset_type = utf-8
        charset_table = (null)
        wordforms = /var/lib/sphinxsearch/data/index/wordforms_main.txt
}
```

**Comment by gjn** [comment]
Regarding your example `Ch. des Prés-du-Dimanche` I don't fully understand. `Ch.` is not in the wordforms, but still gives back 1 result.
 And `Ch des Prés-du-Dimanche` seems to work (same results as in #150), as `ch` is in the wordforms.

**Comment by ponceta** [comment]
The "." is here to check that ch is really transformed into chemin by the wordforms

You can try it with "chemin. des pres du dimanche" works when "ch. des pres du dimanche" does not as expected.

I never said it was logic or straight forward. ;)

**Comment by gjn** [comment]
I don't understand.

**Comment by ltclm** [comment]
you can use the console for debugging
search for ``rte de la gare Marnand``
p.e. rte will be converted to route (as excpected by the wordforms)
```bash
$ search -i address "rte de la gare Marnand"
Sphinx 2.1.6-id64-release (rel21-r4571)
Copyright (c) 2001-2014, Andrew Aksyonoff
Copyright (c) 2008-2014, Sphinx Technologies Inc (http://sphinxsearch.com)

using config file '/etc/sphinxsearch/sphinx.conf'...
index 'address': query 'rte de la gare Marnand ': returned 0 matches of 0 total in 0.000 sec

words:
1. 'route': 94988 documents, 95245 hits
2. 'de': 182624 documents, 197442 hits
3. 'la': 93719 documents, 111856 hits
4. 'gare': 3809 documents, 3809 hits
```

If you search for ``route de la gare Marnand``, route will be converted to rue by wordforms.
```bash
$ search -i address "route de la gare Marnand"
Sphinx 2.1.6-id64-release (rel21-r4571)
Copyright (c) 2001-2014, Andrew Aksyonoff
Copyright (c) 2008-2014, Sphinx Technologies Inc (http://sphinxsearch.com)

using config file '/etc/sphinxsearch/sphinx.conf'...
index 'address': query 'route de la gare Marnand ': returned 31 matches of 31 total in 0.000 sec

displaying matches:
1. document=1586960, weight=4570, feature_id=280061210_0, detail=route de la gare  1524 marnand 5831 valbroye ch vd, label=Route de la Gare  <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001313321001232, geom_st_box2d=BOX(558667 178649,558667 178649), rank=6, x=178649.000000, y=558667.000000, lat=46.757736, lon=6.897625, num=0
2. document=1586961, weight=4570, feature_id=280096134_0, detail=route de la gare  1524 marnand 5831 valbroye ch vd, label=Route de la Gare  <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001312121202322, geom_st_box2d=BOX(558430 178748,558430 178748), rank=6, x=178748.000000, y=558430.000000, lat=46.758614, lon=6.894514, num=0
3. document=1586962, weight=4570, feature_id=280092942_0, detail=route de la gare  1524 marnand 5831 valbroye ch vd, label=Route de la Gare  <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001312132013122, geom_st_box2d=BOX(558470 178735,558470 178735), rank=6, x=178735.000000, y=558470.000000, lat=46.758499, lon=6.895039, num=0
4. document=1586963, weight=4570, feature_id=824051_0, detail=route de la gare  1524 marnand 5831 valbroye ch vd, label=Route de la Gare  <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001313321331312, geom_st_box2d=BOX(558691 178628,558691 178628), rank=6, x=178628.000000, y=558691.000000, lat=46.757549, lon=6.897941, num=0
5. document=1586964, weight=4570, feature_id=280002521_0, detail=route de la gare  1524 marnand 5831 valbroye ch vd, label=Route de la Gare  <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001313200300223, geom_st_box2d=BOX(558531 178693,558531 178693), rank=6, x=178693.000000, y=558531.000000, lat=46.758125, lon=6.895841, num=0
6. document=1586965, weight=4570, feature_id=280023039_0, detail=route de la gare  1524 marnand 5831 valbroye ch vd, label=Route de la Gare  <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001032211133103, geom_st_box2d=BOX(557460 179168,557460 179168), rank=6, x=179168.000000, y=557460.000000, lat=46.762329, lon=6.881779, num=0
7. document=1586966, weight=4570, feature_id=280092938_0, detail=route de la gare  1524 marnand 5831 valbroye ch vd, label=Route de la Gare  <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001312132013122, geom_st_box2d=BOX(558470 178735,558470 178735), rank=6, x=178735.000000, y=558470.000000, lat=46.758499, lon=6.895039, num=0
8. document=1586967, weight=4570, feature_id=280092941_0, detail=route de la gare  1524 marnand 5831 valbroye ch vd, label=Route de la Gare  <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001312132013122, geom_st_box2d=BOX(558470 178735,558470 178735), rank=6, x=178735.000000, y=558470.000000, lat=46.758499, lon=6.895039, num=0
9. document=1586968, weight=4570, feature_id=824036_0, detail=route de la gare 1 1524 marnand 5831 valbroye ch vd, label=Route de la Gare 1 <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001331113313111, geom_st_box2d=BOX(558750 178546,558750 178546), rank=6, x=178546.000000, y=558750.000000, lat=46.756813, lon=6.898721, num=1
10. document=1586969, weight=4570, feature_id=824052_0, detail=route de la gare 2 1524 marnand 5831 valbroye ch vd, label=Route de la Gare 2 <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001313332013323, geom_st_box2d=BOX(558705 178616,558705 178616), rank=6, x=178616.000000, y=558705.000000, lat=46.757442, lon=6.898126, num=2
11. document=1586970, weight=4570, feature_id=824035_0, detail=route de la gare 3 1524 marnand 5831 valbroye ch vd, label=Route de la Gare 3 <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001331112112331, geom_st_box2d=BOX(558717 178558,558717 178558), rank=6, x=178558.000000, y=558717.000000, lat=46.756920, lon=6.898288, num=3
12. document=1586971, weight=4570, feature_id=824034_0, detail=route de la gare 5 1524 marnand 5831 valbroye ch vd, label=Route de la Gare 5 <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001331110211011, geom_st_box2d=BOX(558704 178579,558704 178579), rank=6, x=178579.000000, y=558704.000000, lat=46.757111, lon=6.898116, num=5
13. document=1586972, weight=4570, feature_id=3164110_0, detail=route de la gare 8 1524 marnand 5831 valbroye ch vd, label=Route de la Gare 8 <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001313321001232, geom_st_box2d=BOX(558667 178649,558667 178649), rank=6, x=178649.000000, y=558667.000000, lat=46.757736, lon=6.897625, num=8
14. document=1586973, weight=4570, feature_id=824049_0, detail=route de la gare 12 1524 marnand 5831 valbroye ch vd, label=Route de la Gare 12 <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001313211030122, geom_st_box2d=BOX(558613 178702,558613 178702), rank=6, x=178702.000000, y=558613.000000, lat=46.758209, lon=6.896914, num=12
15. document=1586974, weight=4570, feature_id=824048_0, detail=route de la gare 18 1524 marnand 5831 valbroye ch vd, label=Route de la Gare 18 <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001313021022212, geom_st_box2d=BOX(558546 178756,558546 178756), rank=6, x=178756.000000, y=558546.000000, lat=46.758690, lon=6.896032, num=18
16. document=1586975, weight=4570, feature_id=824047_0, detail=route de la gare 20 1524 marnand 5831 valbroye ch vd, label=Route de la Gare 20 <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001313020100223, geom_st_box2d=BOX(558531 178766,558531 178766), rank=6, x=178766.000000, y=558531.000000, lat=46.758781, lon=6.895834, num=20
17. document=1586976, weight=4570, feature_id=824046_0, detail=route de la gare 22 1524 marnand 5831 valbroye ch vd, label=Route de la Gare 22 <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001312113311230, geom_st_box2d=BOX(558513 178781,558513 178781), rank=6, x=178781.000000, y=558513.000000, lat=46.758915, lon=6.895597, num=22
18. document=1586977, weight=4570, feature_id=3164112_0, detail=route de la gare 23 1524 marnand 5831 valbroye ch vd, label=Route de la Gare 23 <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001313201032321, geom_st_box2d=BOX(558555 178697,558555 178697), rank=6, x=178697.000000, y=558555.000000, lat=46.758160, lon=6.896155, num=23
19. document=1586978, weight=4570, feature_id=824033_0, detail=route de la gare 23 1524 marnand 5831 valbroye ch vd, label=Route de la Gare 23 <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001313201102200, geom_st_box2d=BOX(558560 178705,558560 178705), rank=6, x=178705.000000, y=558560.000000, lat=46.758232, lon=6.896220, num=23
20. document=1586979, weight=4570, feature_id=824045_0, detail=route de la gare 24 1524 marnand 5831 valbroye ch vd, label=Route de la Gare 24 <b>1524 Marnand</b>, origin=address, geom_quadindex=021221001312113012311, geom_st_box2d=BOX(558497 178793,558497 178793), rank=6, x=178793.000000, y=558497.000000, lat=46.759022, lon=6.895387, num=24

words:
1. 'rue': 108828 documents, 108829 hits
2. 'de': 182624 documents, 197442 hits
3. 'la': 93719 documents, 111856 hits
4. 'gare': 3809 documents, 3809 hits
5. 'marnand': 648 documents, 648 hits
```

**Comment by ponceta** [comment]
Wordforms has undesired circular referrences, I'll fix this

**Comment by ponceta** [comment]
The goal behind was to obtain more results using same keywords (like route ~ avenue ~ boulevard)

**Comment by ponceta** [comment]
This can be closed as soon as #205 is merged and swissearch indices deployed
